src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -98,13 +98,24 @@ public function __construct(array $options = [])
         } elseif (isset($_SERVER['argv']) && class_exists(ArgvInput::class)) {
             $this->options = $options;
             $this->getInput();
+            $inputEnv = $_SERVER[$envKey] ?? null;
+            $inputDebug = $_SERVER[$debugKey] ?? null;
         }
 
         if (!($options['disable_dotenv'] ?? false) && isset($options['project_dir']) && !class_exists(MissingDotenv::class, false)) {
             (new Dotenv($envKey, $debugKey))
                 ->setProdEnvs((array) ($options['prod_envs'] ?? ['prod']))
                 ->usePutenv($options['use_putenv'] ?? false)
-                ->bootEnv($options['project_dir'].'/'.($options['dotenv_path'] ?? '.env'), 'dev', (array) ($options['test_envs'] ?? ['test']), $options['dotenv_overload'] ?? false);
+                ->bootEnv($options['project_dir'].'/'.($options['dotenv_path'] ?? '.env'), 'dev', (array) ($options['test_envs'] ?? ['test']), $overrideExistingVars = $options['dotenv_overload'] ?? false);
+            if ($overrideExistingVars) {
+                if (($inputEnv ?? $_SERVER[$envKey]) !== $_SERVER[$envKey]) {
```

Comment:
Maybe more explicit:
```suggestion
                if (isset($inputEnv) && $inputEnv !== $_SERVER[$envKey]) {
```

Reply:
Works for me.